src/CP/core/cpmodification.jl

```diff
@@ -15,6 +15,19 @@ function merge!(prunedDomains::CPModification, newPrunedDomains::CPModification)
     end
 end
 
+function updateChildren!(prunedDomains::CPModification, x::Union{IntVar, BoolVar}, pruned::Union{Array{Int}, Array{Bool}, BitArray})
+    for y in x.children
+        childrenPruned = childrenValue.([y], pruned)
+        if haskey(prunedDomains, y.id)
+            prunedDomains[y.id] = vcat(prunedDomains[y.id], Array(childrenPruned))
+        else
+            prunedDomains[y.id] = Array(childrenPruned)
+        end
+    end
+    return
+end
```

Comment:
Nice spot ! How we couldn't think of this previously, this looks so obvious now ...